### DISCUSSION TITLE
http listener with Digest Authentication returns 401 when query string (URL parameters) are passed with Get

### DISCUSSION THREAD

**Discussion by humairnoor** [nextgenhealthcare/connect#5538] [Q&A]
Hi,
I have created http listener, requirement is to use "Digest Authentication", it works perfect with simple endpoint Get request but when I pass query string / url parameters  in "Get" request , I received "401 unauthorized" . I will really appreciate any guidance in this regard.

My Mirth version: 3.5.0

<h2>HTTP ERROR: 401</h2>
	<p>Problem accessing /patient/address. Reason:
		<pre>    Unauthorized</pre>
	</p>
	<hr /><a href="http://eclipse.org/jetty">Powered by Jetty:// 9.3.14.v20161028</a>

**Comment by pacmano1** [comment]
Post your channel.

**Reply by humairnoor** [reply]
Thanks @pacmano1 , attached is my channel:
[HTTP Listenr with Digest Authentication.zip](https://github.com/nextgenhealthcare/connect/files/10047817/HTTP.Listenr.with.Digest.Authentication.zip)

**Reply by pacmano1** [reply]
And post the sender also please.

**Reply by humairnoor** [reply]
Here is my postman collection I am using to call my channel, Thanks a lot !
[HTTP Listenr with Digest Authentication.zip](https://github.com/nextgenhealthcare/connect/files/10048045/HTTP.Listenr.with.Digest.Authentication.zip)

**Reply by pacmano1** [reply]
No Postman collection in that zip

**Comment by humairnoor** [comment]
Really sorry, missed the collection: here it is:
[Digest -mirth calls.postman_collection.zip](https://github.com/nextgenhealthcare/connect/files/10048346/Digest.-mirth.calls.postman_collection.zip)

**Reply by pacmano1** [reply]
Not an expert on digest - but based on https://stackoverflow.com/questions/5288150/is-digest-authentication-possible-with-jquery/5288679#5288679, it seems your first request is expected to fail and you will receive back an authentication token in a header that you need to use on the next call.
